src/writers.scala

```diff
@@ -41,9 +41,13 @@ trait IdentWriter
     { val c = name(0); Symbol.is_ascii_letter(c) || c == '_' } &&
     name.forall(c => Symbol.is_ascii_letter(c) || Symbol.is_ascii_digit(c) || c == '_')
 
+  val suffix = "_"
+
   def escape(name: String): String =
     if (name.containsSlice("|}")) Exn.error("Bad name: " + Library.quote(name))
-    else "{|" + name + "|}"
+    else if ({val lastChar = name.charAt(name.length - 1); lastChar.isDigit || lastChar == '_'})
+        "{|" + name + suffix + "|}"
+      else "{|" + name + "|}"
```

Comment:
Why is this needed?

Reply:
lambdapi now rejects idents ending with numbers if they have leading zeros, and at some locations idents that must be escaped.
I have not yet well understood how this latter restriction works (the symbols may have escaped-only idents for example) so this is a cherry-pick of what I have found to cause problems : idents of shape `:ddd` (d: digit) used in lambda abstractions